onnxruntime/core/providers/mkldnn/mkldnn_execution_provider.cc

```diff
@@ -130,7 +132,20 @@ bool MKLDNNExecutionProvider::UseSubgraph(const onnxruntime::GraphViewer& graph_
       FP16_graph = node->InputDefs()[0]->Type()->find("16") != std::string::npos;
   }
 
-  if (FP16_graph) {
+  for (auto node_index = 0; node_index < graph_viewer.MaxNodeIndex(); node_index++) {
+    auto node = graph_viewer.GetNode(node_index);
+    if (node == nullptr) {
+      node_index++;
+      continue;
+    }
+    auto op_it = mkldnn_ops_.find(node->OpType());
+    if (op_it != mkldnn_ops_.end()) {
+      mkldnn_nodes_in_the_graph = true;
+      break;
+    }
+  }
+
+  if (FP16_graph || !mkldnn_nodes_in_the_graph) {
```

Comment:
not quite understanding below logic. why is UseSubGraph calling the default GetCapability() method? 
it's a little confusing because kernel_registries will be empty and effectively it will return empty capabilities list.
why not just do that directly?

Reply:
discussed offline about how to refactor the logic

Reply:
Changed the logic in UseSubgraph. 
UseSubgraph now just returns true or false.
GetCapability() calls IExecutionProvider::GetCapability if UseSubgraph returns false.